### PR TITLE
workflows/tests: use `container`

### PR DESCRIPTION
**Pull request by carlocab**
This should help simplify this workflow a bit.

### CODE REVIEW

**Comment by Bo98** [comment]
setup-ruby is not supported in containers. The expectation is you install it yourself.

The reason being that containers won't necessarily match the host OS, and the toolcache is from the host OS.

**Comment by MikeMcQuaid** [comment]
> I think the easiest solution is to split the rspec stuff into their own job without a container. No `bundle install` should be needed for regular command testing - it's only for rspec.

Agreed 👍🏻

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -4,145 +4,83 @@ on:
...
   pull_request:
 jobs:
+  rspec:
+    name: RSpec
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@main
+
+      - name: Set up Ruby
+        if: runner.os == 'Linux'
```
Comment (MikeMcQuaid) [inline .github/workflows/tests.yml:14]:
```suggestion
```
as this only runs on Linux

**Comment by carlocab** [comment]
Seems to be erroring at

https://github.com/Homebrew/homebrew-test-bot/blob/7d0f6809f7da432923088ba6b5fcb8ec62765ae4/lib/tests/formulae.rb#L333

on Linux. Not entirely clear on why...

**Comment by MikeMcQuaid** [comment]
@carlocab Need to run it from another directory so it's not trying to create a symlink across multiple devices/mount points. `mount -a` will probably help here.

**Comment by carlocab** [comment]
> @carlocab Need to run it from another directory so it's not trying to create a symlink across multiple devices/mount points. `mount -a` will probably help here.

Thanks; fixed now. Required tests need updating though.

**Comment by MikeMcQuaid** [comment]
Thanks again @carlocab!

**Pull request merged**